File: game/utilities/WordManager.java
package game.utilities;

import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.FileReader;
import java.io.FileWriter;

public class WordManager {

    public static void addWord(String word) {
        try{
            BufferedWriter output = new BufferedWriter(new FileWriter("game/utilities/hangmanWords.txt", true));
            output.newLine();
            output.append(word);
            output.close();
        } catch (java.io.IOException ex) {
            System.out.println("Missing hangmanWords.txt");
        }
    }

    public static void listWords() {
        try {
            BufferedReader br = new BufferedReader(new FileReader("game/utilities/hangmanWords.txt"));
            String line = "";
            while((line = br.readLine()) != null) {
                System.out.println(line);
            }
        } catch (java.io.IOException ex) {
            System.out.println("Missing hangmanWords.txt");
        }
    }

}

File: game/utilities/word_manager.c
#include <stdlib.h>
#include <stdio.h>

int add_word(char *word){
	FILE *fp;

	fp = fopen("game/utilities/hangmanWords.txt", "a");
	if(!fp){
		fprintf(stderr, "Could not open hangmanWords.txt\n");
		return 1;
	}

	fputs(word, fp);
	fputc('\n', fp);

	fclose(fp);

	return 0;
}

int list_words(){
	FILE *fp;
	int size;
	char *buffer;

	fp = fopen("game/utilities/hangmanWords.txt", "r");
	if(!fp){
		fprintf(stderr, "Could not open hangmanWords.txt\n");
		return 1;
	}

	fseek(fp, 0, SEEK_END);
	size = ftell(fp);
	fseek(fp, 0, SEEK_SET);
	buffer = calloc((size + 1), sizeof(char));
	if(!buffer){
		fprintf(stderr, "Out of memory\n");
		return 1;
	}
	fread(buffer, sizeof(char), size, fp);
	fclose(fp);

	fputs(buffer, stdout);
	fputc('\n', stdout);

	free(buffer);

	return 0;
}
